edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/agentStart.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/sh
+
+###############################################################################
+# Set up EdgeAgent to run as a non-root user at runtime, if allowed.
+# 
+# If this script is started as root:
+#  1. It reads the EDGEAGENTUSER_ID environment variable, default UID=1000.
+#  2. If the User ID does not exist as a user, create it.
+#  3. If "StorageFolder" env variable exists, use as basepath, else use /tmp
+#     Do same for backuppath
+#  4. If basepath/edgeAgent exists, make sure all files are owned by EDGEAGENTUSER_ID
+#     Do same for backuppath/edgeAgent_backup
+#  5. Make sure file specified by IOTEDGE_MANAGEMENTURI is owned by EDGEAGENTUSER_ID
+#  6. Set user id as EDGEAGENTUSER_ID.
+# then start Edge Agent.
+#
+# This preserves backwards compatibility with earlier versions of edgeAgent and
+# allows some flexibility in the assignment of the edgeagent user id. The default 
+# is UID 1000.
+#
+# A user is created because at this time DotNet Core 2.x and 3.x can only install
+# trust bundles into system stores or user stores.  We choose a user store in
+# the code, so a writeable user directory is required.
+###############################################################################
+echo "$(date --utc +"%Y-%m-%d %H:%M:%S %:z") Starting Edge Agent"
+
+TARGET_UID="${EDGEAGENTUSER_ID:-1000}"
+cuid=$(id -u)
+
+if [ $cuid -eq 0 ]
+then
+  # which user creation tool to use.
+  usercreate=useradd
+  if cat /etc/os-release | grep -i alpine > /dev/null
+  then
+    # Alpine only has "adduser"
+    usercreate=adduser
+  fi
+
+  # Create the agent user id if it does not exist
+  if ! getent passwd "${TARGET_UID}" >/dev/null
+  then
+    echo "Creating UID ${TARGET_UID} as agent${TARGET_UID}"
+    if [ $usercreate = useradd ]
```

Comment:
Nit pick, why not just check the alpine here and choose the command accordingly?

Reply:
I took @arsing 's suggestion (use `command -v`), because it's more elegant than what I did, and is the only place where the commands differ between Alpine and Bionic (our two agent bases).